crates/core/tedge_agent/src/operation_workflows/actor.rs

```diff
@@ -465,6 +482,71 @@ impl WorkflowActor {
             _ => Err(CommandTopicError::InvalidCommandTopic),
         }
     }
+
+    fn process_iterate(
+        &mut self,
+        state: GenericCommandState,
+        target_json_path: String,
+        handlers: IterateHandlers,
+    ) -> Result<GenericCommandState, IterationError> {
+        let Some(json_path) = GenericCommandState::extract_path(&target_json_path) else {
+            return Err(IterationError::InvalidPathExpression(target_json_path));
+        };
```

Comment:
This should be done while parsing the workflow definition in `impl TryFrom<TomlExitHandlers> for IterateHandlers`

Reply:
Resolved by 1012f41c0069b302fb54aa249149fdee9ab20071